climada/hazard/relative_cropyield.py

```diff
@@ -247,18 +252,15 @@ def set_rel_yield_to_int(self, hist_mean):
         idx = np.where(hist_mean != 0)[0]
 
         # initialize new hazard_matrix
-        hazard_matrix = np.empty(self.intensity.shape)
-        hazard_matrix[:, :] = np.nan
+        hazard_matrix = np.zeros(self.intensity.shape, dtype=np.float32)
 
         # compute relative yield for each event:
         for event in range(len(self.event_id)):
```

Comment:
Why first make a non-sparse matric and then convert it to a sparse matrix? This seems not very efficient...

Reply:
Would it make more sense to create a copy of self.intensity?

Reply:
> Why first make a non-sparse matric and then convert it to a sparse matrix? This seems not very efficient...

which lines are you referring to exactly?